server/filestore.go

```diff
@@ -2975,12 +2975,14 @@ func (fs *fileStore) Compact(seq uint64) (uint64, error) {
 
 	smb.mu.Lock()
 	for mseq := smb.first.seq; mseq < seq; mseq++ {
-		if _, rl, _, err := smb.slotInfo(int(mseq - smb.cache.fseq)); err != nil {
-			smb.bytes -= uint64(rl)
+		sm, _ := smb.cacheLookupWithLock(mseq)
+		if sm != nil && smb.msgs > 0 {
```

Comment:
Can you make it a one line? `if sm, _ := smb.cacheLookupWithLock(mseq); sm != nil && smb.msgs > 0 {`